tests/integration/workload-objects-integration-test.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# This script is expected to be called from the root folder of Connaisseur
+
+# List of Workload Objects
+woList=("CronJob" "DaemonSet" "Deployment" "Job" "Pod" "ReplicaSet" "ReplicationController" "StatefulSet")
+
+echo '** Prepare Connaisseur config'
+yq eval-all --inplace 'select(fileIndex == 0) * select(fileIndex == 1)' helm/values.yaml tests/integration/preconfig-update.yaml
+echo '>> Config set'
+
+echo -e '\n** Install Connaisseur'
+helm install connaisseur helm --atomic --create-namespace --namespace connaisseur || { echo '>> Failed to install Connaisseur'; exit 1; }
+echo '>> Successfully installed Connaisseur'
+
+for wo in "${woList[@]}"; do
+    export KIND=${wo}
+    echo -e "\n##################################\n** Kubernetes Resource: ${KIND}"
+
+    export APIVERSION=$(kubectl api-resources | awk -v KIND=${KIND} '{ if($NF == ""KIND"") print $(NF-2);}')
+    echo "** API Version: ${APIVERSION}"
+
+    # UNSIGNED
+    export TAG=unsigned
+
+    echo -e "\n** Render '${KIND}' using '${APIVERSION}' and '${TAG}' image"
+    envsubst < tests/integration/workload-objects/${KIND}.yaml | cat
+
+    echo -e "\n** Deploy '${KIND}' using '${APIVERSION}' and '${TAG}' image"
+    envsubst < tests/integration/workload-objects/${KIND}.yaml | kubectl apply -f - >output.log 2>&1 || true
+
+    if [[ ! "$(cat output.log)" =~ 'Unable to find signed digest for image docker.io/securesystemsengineering/testimage:unsigned.' ]]; then
```

Comment:
It would probably be sufficient to deploy a valid signed image without error, wouldn't it? But as long as we don't run into timing issues, I don't mind testing more :)

Reply:
that is what I thought and it seems that the k3s never fails with timing issues